### PR TITLE
[testing] Better cross-platform testing

### PR DESCRIPTION
**Pull request by davezarzycki**
1) Don't force ObjC interop just to enable gratuitous `@objc deinit`.
2) Prefer `-enable-objc-interop` over `REQUIRES: objc_interop`.

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please smoke test

**Review by DougGregor — APPROVED** [review]
These all LGTM

File: test/SILOptimizer/sil_combine_objc.sil [hunk 1]
```diff
@@ -1,5 +1,4 @@
-// RUN: %target-sil-opt -enable-sil-verify-all %s -sil-combine | %FileCheck %s
-// REQUIRES: objc_interop
+// RUN: %target-sil-opt --enable-objc-interop -enable-sil-verify-all %s -sil-combine | %FileCheck %s
```
Comment (compnerd) [inline test/SILOptimizer/sil_combine_objc.sil:1]: Is this supposed to be `-enable-objc-interop`?

File: test/SILOptimizer/sil_combine_objc_ossa.sil [hunk 2]
```diff
@@ -1,5 +1,4 @@
-// RUN: %target-sil-opt -enable-sil-verify-all %s -sil-combine | %FileCheck %s
-// REQUIRES: objc_interop
+// RUN: %target-sil-opt --enable-objc-interop -enable-sil-verify-all %s -sil-combine | %FileCheck %s
```
Comment (compnerd) [inline test/SILOptimizer/sil_combine_objc_ossa.sil:1]: Similar

**Review by compnerd — COMMENTED** [review]
Thanks for doing this - getting better coverage is important, and this definitely helps.

**Comment by davezarzycki** [comment]
Fixed. Those were `XFAIL: *` so that's why I didn't catch it.

@swift-ci please smoke test

**Comment by davezarzycki** [comment]
@swift-ci please smoke test linux

**Pull request merged**